candidate/server.py

```
@@ -1,23 +1,53 @@
+    try:
+        os.unlink(temp_path)
+    except OSError:
+        pass
 
+    # 4) Совмещаем с вакансией и возвращаем результат
+    try:
+        result = process_json(resume_dict, vacancies)
         return result
     except Exception as e:
-        raise HTTPException(status_code=400, detail=f"Ошибка обработки JSON: {str(e)}")
+        raise HTTPException(status_code=400, detail=f"Ошибка при обработке данных: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)